File: wrappers/dotnet/indy-sdk-dotnet-test/EnvironmentUtils.cs
using System;
using System.IO;

namespace Hyperledger.Indy.Sdk.Test
{
    static class EnvironmentUtils
    {
        public static string GetTestPoolIP()
        {
            var testPoolIp = Environment.GetEnvironmentVariable("TEST_POOL_IP");
            return testPoolIp != null ? testPoolIp : "127.0.0.1";
        }

        public static string GetUserHomePath()
        {
            return Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE")); //TODO: Does this work on non-Windows platforms?
        }

        public static string GetIndyHomePath()
        {
            return Path.Combine(GetUserHomePath(), ".indy");
        }

        public static string getIndyHomePath(string filename)
        {
            return GetIndyHomePath() + filename;
        }

        public static string GetTmpPath()
        {
            return Path.Combine(Path.GetTempPath(), "indy");
        }

        public static string GetTmpPath(string filename)
        {
            return GetTmpPath() + filename;
        }

    }
}

File: wrappers/android/src/androidTest/java/org/hyperledger/indy/sdk/utils/EnvironmentUtils.java
package org.hyperledger.indy.sdk.utils;

import androidx.test.platform.app.InstrumentationRegistry;

public class EnvironmentUtils {
    static String getTestPoolIP() {
        // TODO this is not as usefull on android. Could be configured through buildConfigField
        //  on test apk. DSL does not allow that at the moment.
        String testPoolIp = System.getenv("TEST_POOL_IP");
        return testPoolIp != null ? testPoolIp : "10.0.0.2";
    }

    public static String getIndyHomePath() {
        return InstrumentationRegistry.getInstrumentation().getContext().getFilesDir().getAbsolutePath() + "/indy_client/";
    }

    public static String getIndyHomePath(String filename) {
        return getIndyHomePath() + filename;
    }

    public static String getTmpPath() {
        return InstrumentationRegistry.getInstrumentation().getContext().getCacheDir().getAbsolutePath() + "/tmp/";
    }

    public static String getTmpPath(String filename) {
        return getTmpPath() + filename;
    }
}
